update.py

```diff
@@ -219,6 +226,65 @@ def copy_build(site):
         shutil.rmtree(olddir)
 
 
+def copy_and_keep_build(site):
+    """
+    Copies each site into the target location and keep last "n" builds as backups
+    """
+    global error_count
+    for wiki in ALL_WIKIS:
+        if site=='common':
+            continue
+        if site is not None and site != wiki:
+            continue
+        debug('copy: %s' % wiki)
+        targetdir = os.path.join(args.destdir, wiki)
+
+        debug("Creating temporary folders")
+        previousdir = os.path.join(args.backupdestdir)
+        debug('Recreating %s' % previousdir )
+        if not os.path.exists(previousdir):
+            os.mkdir(previousdir)
+    
+        olddir = os.path.join(previousdir, str(building_time) )
+        debug('Recreating %s' % olddir )
+        if not os.path.exists(olddir):
+            os.mkdir(olddir)
+
+        if os.path.exists(targetdir):
+            debug('Moving %s into %s' % (targetdir,olddir) )
+            shutil.move(targetdir, olddir)
+
+        sourcedir='./%s/build/html/' % wiki
+
+        html_moved_dir = os.path.join(args.destdir, 'html')
+        try:
+            subprocess.check_call(['mv', sourcedir, html_moved_dir])
+            # Rename move! (single move to html/* failed)
+            subprocess.check_call(['mv', html_moved_dir ,targetdir])
+            debug("Moved to %s" % targetdir)
+        except:
+            error("FAIL moving output to %s" % targetdir)
+        finally:
+            debug("Creating a backup in %s" % olddir)
+            subprocess.check_call(['cp', '-r', targetdir ,olddir])
```

Comment:
This `check_call` does seem to be whole-cloth new.  Please use a Python library to do this one.  Probably `shutil.copytree`

Reply:
shutil.copytree() is slightly different from python 2.7 to 3.8. So, I used the previous indication distutils.dir_util lib.